index.js

```diff
@@ -19,23 +20,35 @@ const InterestingVariables = [
 ]
 
 function findVcvarsall() {
+    // use vswhere
+    let path = child_process.execSync('vswhere -products * -latest -prerelease -find **/Auxiliary/Build/vcvarsall.bat').toString().trim()
+    if (path && fs.existsSync(path)) {
+        return path
+    }
```

Comment:
Hm... What if `vswhere` is not actually installed in the system, or it fails to find the file? It seems it will throw an exception, and then the probing approach will not be executed at all, and we won't look for `vcbuildtools.bat` either.

I think it would be better for extract this into a helper function like `findWithVswhere()` which accepts a string like `"Auxiliary/Build/vcvarsall.bat"`, tries to run vswhere, handles exceptions, checks the returned path for existence, and returns either a path or `null`. It can also pring some debug logs if it fails to find the file.

Reply:
Fixed in https://github.com/ilammy/msvc-dev-cmd/pull/11/commits/d4aa56d8b3f0fefb0ac005cf0fefa30f8ab77af5